modules/images/webp-uploads/load.php

```diff
@@ -704,3 +704,71 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
 	return rest_ensure_response( $data );
 }
 add_filter( 'rest_prepare_attachment', 'webp_uploads_update_rest_attachment', 10, 3 );
+
+/**
+ * Returns the attachment sources array ordered by filesize.
+ *
+ * @since n.e.x.t
+ *
+ * @param int    $attachment_id The attachment ID.
+ * @param string $size          The attachment size.
+ *
+ * @return array The attachment sources array.
+ */
+function webp_uploads_get_attachment_sources( $attachment_id, $size = 'thumbnail' ) {
+	// Check for the sources attribute in attachment metadata.
+	$metadata = wp_get_attachment_metadata( $attachment_id );
+
+	// Return full image size sources.
+	if ( 'full' === $size && ! empty( $metadata['sources'] ) ) {
+		return $metadata['sources'];
+	}
+
+	// Return the resized image sources.
+	if ( ! empty( $metadata['sizes'][ $size ]['sources'] ) ) {
+		return $metadata['sizes'][ $size ]['sources'];
+	}
+
+	// Return an empty array if no sources found.
+	return array();
+}
+
+/**
+ * Filters on `webp_uploads_content_image_mimes` to generate the available mime types for an attachment
+ * and orders them by the smallest filesize first.
+ *
+ * @since n.e.x.t
+ *
+ * @param array  $mime_types    The list of mime types that can be used to update images in the content.
+ * @param int    $attachment_id The attachment ID.
+ * @param string $context       The current context.
+ *
+ * @return array Array of available mime types ordered by filesize.
+ */
+function webp_uploads_get_mime_types_by_filesize( $mime_types, $attachment_id, $context ) {
```

Comment:
The context parameter seems to be redundant for this function. Let's not use it then:

```php
function webp_uploads_get_mime_types_by_filesize( $mime_types, $attachment_id ) {
    ...
}
add_filter( 'webp_uploads_content_image_mimes', 'webp_uploads_get_mime_types_by_filesize', 10, 2 );
```

Reply:
Updated and removed the `$context` parameter